extensions/git/src/commands.ts

```diff
@@ -202,18 +202,35 @@ async function categorizeResourceByResolution(resources: Resource[]): Promise<{
 
 function createCheckoutItems(repository: Repository): CheckoutItem[] {
 	const config = workspace.getConfiguration('git');
-	const checkoutType = config.get<string>('checkoutType') || 'all';
-	const includeTags = checkoutType === 'all' || checkoutType === 'tags';
-	const includeRemotes = checkoutType === 'all' || checkoutType === 'remote';
-
-	const heads = repository.refs.filter(ref => ref.type === RefType.Head)
-		.map(ref => new CheckoutItem(ref));
-	const tags = (includeTags ? repository.refs.filter(ref => ref.type === RefType.Tag) : [])
-		.map(ref => new CheckoutTagItem(ref));
-	const remoteHeads = (includeRemotes ? repository.refs.filter(ref => ref.type === RefType.RemoteHead) : [])
-		.map(ref => new CheckoutRemoteHeadItem(ref));
-
-	return [...heads, ...tags, ...remoteHeads];
+	const checkoutType = config.get<string>('checkoutType') || 'local,remote,tags';
+	const checkoutTypes = checkoutType.trim().split(',').map(type => type.trim());
```

Comment:
You can filter and unique the valid types in advance rather than collecting invalids in the loop.
```suggestion
	const checkoutTypeString = config.get<string>('checkoutType') || 'local,remote,tags';
	const checkoutTypes = Array.from(new Set(checkoutTypeString.trim().split(',').map(type => type.trim()).filter(type => ['local','remote','tags'].indexOf(type) > 0)));
```